2 - cluster organisations and link them to the STM topics/preprocessing.py

```diff
@@ -0,0 +1,273 @@
+from text_preprocessing import preprocess_text
+from text_preprocessing import to_lower, remove_number, remove_itemized_bullet_and_numbering, remove_special_character, remove_punctuation, remove_whitespace, normalize_unicode, check_spelling, remove_url, remove_email, remove_phone_number, remove_ssn, remove_whitespace
+from nltk.tokenize import word_tokenize, sent_tokenize
+from nltk.corpus import stopwords
+from nltk.stem import WordNetLemmatizer
+import re
+import pandas as pd
+import os.path
+import json
+import string
+from pathlib import Path
+
+path = Path(os.getcwd())
+data_dir = os.path.join(path.parents[0], "data-files")
+
+# get data (full body texts of documents)
+with open(os.path.join(data_dir, "data.json")) as f:
+    data = json.load(f)
+
+# get data (the structured data which Malte processed into lines with metadata)
+with open(os.path.join(data_dir, "studies_on_water_scraped.json")) as sf:
+    raw_data = json.load(sf)
+
+with open(os.path.join(data_dir, "ngram_replacements.json")) as nf:
+    ngram_replacements = json.load(nf)
+
+# define function to lookup correct ID for document in studies_on_water_scraped.json
+# before this, I was using the INDEX of the document in the JSON array of this file as its ID.
+def lookup_correct_docid(old_key):
+    global raw_data
+    return raw_data[int(old_key)]['meta']['id']
+
+def replace_all(text, dic):
+    for i, j in dic.items():
+        text = text.replace(i, j)
+    return text
+
+# stopwords (niche irrelevant characters)
+irrelevant_tokens = ['et', 'al.', 'x', 'pdf', 'yes', 'abbrev','fe',
+                            'page', 'pp', 'p', 'er', 'doi', 'b', 'c', 'd', 'e',
+                            'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'o', 'q', 'r', 's',
+                            't', 'u', 'v', 'w', 'y', 'z','www', 'com', 'org', 'de', 'dx', 'th', 'ii', 'le']
+
+def preprocess_ner(corpus, stopwords):
```

Comment:
This function seems to be already defined in `1 - .../preprocessing.py`. It would be good to have the function only in a single location.

Reply:
fixed